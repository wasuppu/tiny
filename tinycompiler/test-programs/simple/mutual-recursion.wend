main() {
    int n;

    bool even(int n) {
        if n == 0 {
            return true;
        } else {
            return odd(abs(n) - 1);
        }
    }

    bool odd(int n) {
        if n == 0 {
            return false;
        } else {
            return even(abs(n) - 1);
        }
    }

    int abs(int n) {
        if n < 0 {
            return -n;
        } else {
            return n;
        }
    }

    n = -255;
    print "odd("; print n; print ") = "; println odd(n);
}

