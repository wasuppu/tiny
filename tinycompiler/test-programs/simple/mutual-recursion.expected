odd(-255) = true
